src/Slick/Build.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Slick.Build
+  ( convert
+  , srcToURL
+  , destToSrc
+  , srcToDest
+  , flattenMeta
+  , pruner
+  , EntityFilePath(..)
+  , shakeArgsAlwaysPruneWith
+  ) where
+
+import           Control.Lens
+import           Control.Monad
+import           Data.Aeson
+import           Data.Aeson.Lens
+import           Data.List                  as L (intersperse, sortBy, (\\))
+import           Data.Maybe
+import           Data.Monoid
+import qualified Data.Text                  as T
+import           Development.Shake
+import           Development.Shake.Classes
+import           Development.Shake.FilePath
+import           GHC.Generics               hiding (Meta)
+import           System.Console.GetOpt
+import           System.Directory.Extra     (listFilesRecursive, removeFile)
+import           System.IO.Extra            as IO
+import           Text.Pandoc
+import           Text.Pandoc.Highlighting
+import           Text.Pandoc.Shared
+
+import           Slick.Utils
+
+--------------------------------------------------------------------------------
+
+-- | Generalized version of filepath
```

Comment:
I assume this if for type-directed caching? If so could you describe how it's used? I don't see it in the example, and the docs aren't really clear on it. I feel like this could probably be removed unless we can work it in as a `jsonCache` improvement

Reply:
Yes, to distinguish entities (mostly described by type Post, Project, PostDraft) inside cache. I'll add more description on that.